### ISSUE TITLE
Cygwin awkwardness

### ISSUE THREAD

**Issue by binkley** [python/mypy#548]
Current cygwin python3 is 3.2.5, so I install and use the native Windows Python which is 3.4.2 installed at `C:\Python34` (the default).

After installing mypy via `python setup.py instal`" it installs under `C:\Python34`, as expected, and creates a `C:\Python34\Scripts\mypy` file.  Note there is no `mypy.exe` program.  (There are such programs for `pip.exe` and `easy_install.exe`, would be helpful for mypy to do likewise.)

To run this under cygwin requires an explicit call to `python` (bash turns the script path into a unix-y one, which confuses native Windows python -- not your issue, I mention this for clarity), this:

``` bash
$ export PATH=/cygdrive/c/Python34:$PATH
$ python C:/Python34/Scripts/mypy my-file.py
```

Oddly this complains there is no `enum` module to import, though running `python my-file.py` directly works fine.  Running the same under a native CMD prompt behaves the same.

I'm unsure what about running from a Cygwin bash shell is causing the issue, perhaps it isn't mypy's fault at all but I'm at a loss how to investigate.

**Comment by JukkaL** [comment]
Thanks for reporting this! I'm currently traveling abroad with a busy schedule and I don't have access to my Windows VM, but I'll investigate this once I'm back home (and recovered from jet lag).

**Comment by Igonato** [comment]
Had problem with mypy under Cygwin as well. Found out that using `mypy.bat` as a command instead of `mypy` helps tremendously.

**Comment by JelleZijlstra** [comment]
Maybe we need a documentation change explaining how to use mypy under cygwin?

**Comment by Igonato** [comment]
I don't know, maybe, it's a fairly uncommon setup (Cygwin + win python) but I guess it wouldn't hurt to mention it in the readme.

Also, it can be fixed with a proper mypy file, It looks like this on my machine:

```bash
#!C:\Python36-x64\python.exe
"""Mypy type checker command line tool."""

from mypy.main import main

main(__file__)
```

First, there is no `C:\Python36-x64\python.exe` , the path is `C:\Python36\python.exe` and even if I change it, it doesn't work, since cygwin sends it path in its internal format

```bash
$ mypy .
C:\Python36\python.exe: can't open file '/cygdrive/c/Python36/Scripts/mypy': [Errno 2] No such file or directory
```

Can we change it to `python -m mypy "$@"`? That should work on every system, right? And mypy.bat needs to be changed as well, form `"%~dp0mypy" %*` to ` "-m" "mypy" %*`

**Comment by Igonato** [comment]
^ here is the PR, It should make the scripts more robust. Please check it out when you get the time. Cheers!

**Comment by Igonato** [comment]
Changed my mind:
> It works as `mypy.bat stuff`, it works as `python -m mypy stuff`, that should be enough for Cygwin users.

**Comment by binkley** [comment]
Will try it.  I'm away from home for a couple of weeks, and home is where my Windows box is, so I may appear slow.  :)

**Comment by ethanhs** [comment]
This should no longer be an issue, mypy installs a `mypy.exe` which should handle everything correctly. Feel free to open a new issue if there are further problems.
